### PR TITLE
Change OptionalMessageBodyWriter to produce wildcard

### PR DESCRIPTION
**Pull request by markelliot**
This change is [Reviewable](https://reviewable.io/reviews/palantir/http-remoting/292)

### CODE REVIEW

File: jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/OptionalMessageBodyWriter.java [hunk 1]
```diff
@@ -35,7 +35,7 @@
 import org.glassfish.jersey.message.MessageBodyWorkers;
 
 @Provider
-@Produces(MediaType.APPLICATION_JSON)
+@Produces(MediaType.WILDCARD)
```
Comment (markelliot) [inline jersey-servers/src/main/java/com/palantir/remoting1/servers/jersey/OptionalMessageBodyWriter.java]: the tests already use TEXT_PLAIN return types, but Dropwizard is independently registering something that deals with that already, so this isn't invoked in the existing tests/it's not trivial to add new tests for this.

**Comment by uschi2000** [comment]
:lgtm:

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292#-:-KW-wVM46yRbk_Y_raWF:bnfp4nl)_

**Comment by uschi2000** [comment]
Fine with me, but tests may be unhappy since the http type handling is somewhat brittle.

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292#-:-KW-wcwMAaU23jbQLoX7:b-xgtmnm)_

**Comment by markelliot** [comment]
So I can't even manually get this OptionalMessageBodyWriter to run and OptionalMessageBodyWriterTests actually have an error where the expectation is set to use a 404 mapper for no content, but we'd expect to see a 204. So something is wrong.

**Comment by markelliot** [comment]
@uschi2000 this seems to resolve previously undefined (or underdefined?) behavior, not sure if that's okay with you.

**Comment by uschi2000** [comment]
Cool, thanks! Need to update the spec in the README too, otherwise :LGTM:

---

Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292#-:-KW0bjlF4UGQAed8Y1ak:b-ki5lpo)_

**Comment by markelliot** [comment]
done

---

Review status: 0 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292#-:-KW0d6I9eNz3Ob3m0zs7:bb7f2mf)_

**Comment by markelliot** [comment]
Reviewed 3 of 3 files at r2, 1 of 1 files at r3.
Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292)_

**Comment by uschi2000** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved, some commit checks pending.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/http-remoting/292#-:-KW0drYK_zVMXpSvbbC6:bnfp4nl)_

**Pull request merged**